rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -188,19 +195,65 @@ void Player::play_messages_until_queue_empty(const PlayOptions & options)
   }
 }
 
+bool is_package_found(rosbag2_storage::TopicMetadata topic)
+{
+  std::string package_name = std::get<0>(rosbag2_cpp::extract_type_identifier(topic.type));
+  std::string content;
+  std::string prefix_path;
+  return ament_index_cpp::get_resource("packages", package_name, content, &prefix_path);
+}
+
+std::vector<std::string> Player::update_topics_to_filter_to_play_messages_with_unknwon_types(
+  const std::vector<std::string> & topics_to_filter)
+{
+  std::vector<std::string> topics_to_filter_updated;
+  auto topics = reader_->get_all_topics_and_types();
+  for (const auto & topic : topics) {
+    bool topic_should_be_played = topics_to_filter.size() == 0 ||
+      std::find(
+      topics_to_filter.begin(), topics_to_filter.end(),
+      topic.name) != topics_to_filter.end();
+    if (is_package_found(topic) && topic_should_be_played) {
+      topics_to_filter_updated.emplace_back(topic.name);
+    } else if (!is_package_found(topic)) {
+      std::string package_name = std::get<0>(rosbag2_cpp::extract_type_identifier(topic.type));
+      ROSBAG2_TRANSPORT_LOG_WARN(
+        "All messages on topic '%s' will be ignored because package '%s' is not found.",
+        topic.name.c_str(), package_name.c_str());
+    }
+  }
+  return topics_to_filter_updated;
+}
+
 void Player::prepare_publishers(const PlayOptions & options)
 {
+  auto topics_to_filter = options.topics_to_filter;
+  if (options.ignore_unknown_types) {
+    topics_to_filter = update_topics_to_filter_to_play_messages_with_unknwon_types(
+      options.topics_to_filter);
+    if (topics_to_filter.size() == 0) {
+      ROSBAG2_TRANSPORT_LOG_ERROR("No topic to be sucribed to could be found.");
+      return;
+    }
+  }
```

Comment:
I don't think that's necessary. We apply the filter, get all topics and types and filter then once which of the topics can actually be replayed.

Reply:
alternatively, we filter out all topics which we can publish and pass these in via storage_filters. We can then later on be sure that all topics can be replayed and safely create publishers for it.